proposals/0047-rerouting.md

```diff
@@ -0,0 +1,186 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+- Start Date: 2024-04-17
+- Reference Issues: https://github.com/withastro/roadmap/issues/896
+- Implementation PR: 
+
+# Summary
+
+Programmatic control over routing.
+
+# Background & Motivation
+
+
+As astro grows in popularity, it is being used for more and more use cases. One of these is multitenant websites where a tenant may be given a subdomain. This proposal introduces a powerful primitive that allows astro users to build websites with complex routing requirements. Rather than a limitation, file-based routing is now a sensible default!
+
+# Goals
+
+- Defuse.
+- Make it unnecessary to rely on implicit routing rules
+- Introduce framework-designed 404 and 500 pages
+- Make middleware run for requests that don't match a route (with framework-designed error pages)
+
+# Non-Goals
+
+- Reroute to an external service.
+- Support for `functionPerRoute`
+- fallback rerouting
```

Comment:
I think all of these non-goals are good but could use a bit more explanation for those who might not understand the connections.